velox/functions/lib/Re2Functions.cpp

```diff
@@ -1137,20 +1377,46 @@ class PatternStringIterator {
     } else {
       if (currentChar == '_') {
         charKind_ = CharKind::kSingleCharWildcard;
+        nextStart_ = currentStart_ + 1;
       } else if (currentChar == '%') {
         charKind_ = CharKind::kAnyCharsWildcard;
+        nextStart_ = currentStart_ + 1;
       } else {
         charKind_ = CharKind::kNormal;
+
+        // Unicode.
+        if (currentChar & 0x80) {
+          auto chrOffset =
+              utf8proc_char_length(pattern_.data() + currentStart_);
+          // Skip bad byte if we get utf length < 0.
+          auto numBytes = UNLIKELY(chrOffset < 0) ? 1 : chrOffset;
+          nextStart_ = currentStart_ + numBytes;
+        } else {
+          nextStart_ = currentStart_ + 1;
+        }
       }
-      nextStart_ = currentStart_ + 1;
     }
 
     return true;
   }
 
-  // Start index of the current character.
-  size_t currentStart() const {
-    return currentStart_;
+  // Char at current cursor, since it can be a multibyte character, here we use
+  // a string_view to represent.
+  std::string_view current() const {
+    // Unicode.
+    char startChar = charAt(currentStart_);
+    if (startChar & 0x80) {
+      return std::string_view(
+          pattern_.data() + currentStart_, nextStart_ - currentStart_);
+    }
+
+    // Ascii.
+    // Escaped.
```

Comment:
Is it possible to have a multi-byte escape character?

Reply:
The comments is kinda misleading, by 

```
    // Ascii.
    // Escaped.
```
I mean the following are Ascii code path: the `if` is `escape` path, the `else` path is `normal` path. I made some optimization for the comments.